aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -0,0 +1,49 @@
+#include "ATen/NativeFunctions.h"
+#include "ATen/Dispatch.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+#include <curand.h>
+#include <curand_kernel.h>
+#include <curand_philox4x32_x.h>
+#include <utility>
+
+#include <THC/THCGeneral.h>
+#include <THC/THCHalf.h>
+#include <THC/THCApply.cuh>
+#include <THC/THCNumerics.cuh>
+#include <THC/THCTensorRandom.h>
+
+THCGenerator* THCRandom_getGenerator(THCState* state);
+
+namespace at {
+namespace native {
+
+namespace dist {
+  std::pair<uint64_t, uint64_t> get_philox_seed(Generator *gen) {
```

Comment:
Minor, but I'd prefer `next_philox_seed` or `next_philox_offset` or something like that since this is a mutating method.